npm/webpack-dev-server/src/loader.ts

```diff
@@ -47,20 +48,18 @@ function buildSpecs (projectRoot: string, files: Cypress.Cypress['spec'][] = [])
 
 // Runs the tests inside the iframe
 export default function loader () {
-  const { files, projectRoot } = this._cypress as { files: Cypress.Cypress['spec'][], projectRoot: string }
+  const { files, projectRoot, support } = this._cypress as CypressOptions
 
   return `
+  var loadSupport = ${support ? `() => import(${JSON.stringify(support)})` : `() => Promise.resolve()`}
   var allTheSpecs = ${buildSpecs(projectRoot, files)};
 
-  const { init } = require(${JSON.stringify(require.resolve('./aut-runner'))})
+  var { init } = require(${JSON.stringify(require.resolve('./aut-runner'))})
 
-  const { restartRunner } = init(Object.keys(allTheSpecs)
+  var specLoaders = Object.keys(allTheSpecs)
     .filter(key => allTheSpecs[key].shouldLoad())
-    .map(a => allTheSpecs[a].load())
-  )
+    .map((a) => () => allTheSpecs[a].load())
```

Comment:
Why is having two levels of function wrappers necessary?

Reply:
Could be a reduce one-liner. I am sure this was just to get it done quick. The `map` seems weird - a function that returns a function call? We could just do `map(a => a.load)`.

With reduce, something like this:

```js
Object.entries(allTheSpecs).reduce((acc, curr) => {
  if (curr.shouldLoad()) {
    return acc.concat(curr.load)
  }
  return acc
})
```

Reply:
Yeah I initially wrote this. A reduce would've been fine. Coding a reduce in strings was a bit challenging 😂